google-map-search.html

```diff
@@ -147,13 +169,18 @@
      * Perform a search using for `query` for the search term.
      */
     search: function() {
-      if (this.query && this.map) {
+      if (this.map && (this.query || this.type)) {
         var places = new google.maps.places.PlacesService(this.map);
 
         if (this.types && typeof this.types == 'string') {
           var types = this.types.split(' ');
         }
 
+        if (this.type && typeof this.type == 'string') {
+          var typeArray = this.type.split(' ');
+          var type = typeArray[0];
```

Comment:
Maybe I'm missing something, but is there a reason why `type` needs to be initialized outside a conditional more than any of the others here? Do you want me to init all of them at the top?